snapcraft/internal/project_loader/_env.py

```diff
@@ -89,3 +89,49 @@ def build_env_for_stage(stagedir: str, snap_name: str,
     env.append('PERL5LIB="{0}/usr/share/perl5/"'.format(stagedir))
 
     return env
+
+
+def snapcraft_global_environment(project: project.Project) -> Dict[str, str]:
+    if project.info.name:
+        name = project.info.name
+    else:
+        name = ''
+
+    if project.info.version:
+        version = project.info.version
+    else:
+        version = ''
+
+    if project.info.grade:
+        grade = project.info.grade
+    else:
+        grade = ''
+
+    return {
+        'SNAPCRAFT_ARCH_TRIPLET': project.arch_triplet,
+        'SNAPCRAFT_PARALLEL_BUILD_COUNT': project.parallel_build_count,
+        'SNAPCRAFT_PROJECT_NAME': name,
+        'SNAPCRAFT_PROJECT_VERSION': version,
+        'SNAPCRAFT_PROJECT_GRADE': grade,
+        'SNAPCRAFT_STAGE': project.stage_dir,
+        'SNAPCRAFT_PRIME': project.prime_dir,
+    }
+
+
+def snapcraft_part_environment(
+        part: pluginhandler.PluginHandler) -> Dict[str, str]:
+    return {
+        'SNAPCRAFT_PART_SRC': part.sourcedir,
+        'SNAPCRAFT_PART_BUILD': part.builddir,
+        'SNAPCRAFT_PART_INSTALL': part.installdir,
```

Comment:
if this is the only place `part.installdir` is used, I would rather go the other way around and removing the property definition instead of creating a new one. And if not make this use `part.plugin.<dir>` instead and later on remove the `installdir` property.

Reply:
Done.